modules/ha-cluster/main.tf

```diff
@@ -0,0 +1,894 @@
+  -hcl='tls { defaults { cert_file = "/consul/${var.datacenter}-server-consul-0.pem" }}' \
+  -hcl='tls { defaults { key_file = "/consul/${var.datacenter}-server-consul-0-key.pem" }}' \
+  -hcl='tls { defaults { verify_incoming = true, verify_outgoing = true }}' \
+  -hcl='tls { internal_rpc { verify_server_hostname = true }}' \
+  -hcl='auto_encrypt = {allow_tls = true}' \
+  -hcl='ports { server = 8300, serf_lan = 8301, serf_wan = 8302, https = 8501, grpc = 8502, grpc_tls = 8503 }' \
+%{endif~}
+%{if var.acls~}
+  -hcl='acl {enabled = true, default_policy = "deny", down_policy = "extend-cache", enable_token_persistence = true}' \
+  -hcl='acl = { tokens = { initial_management = "${local.bootstrap_token}", default = "${local.bootstrap_token}" }}' \
+%{endif~}
+%{if var.acls && local.enable_mesh_gateway_wan_federation~}
+  -hcl='acl = { enable_token_replication = true }' \
+%{endif~}
+%{if var.acls && var.replication_token != ""~}
+  -hcl='acl = { tokens = { replication = "${var.replication_token}"}}' \
+%{endif~}
+%{if var.primary_datacenter != ""~}
+  -hcl='primary_datacenter = "${var.primary_datacenter}"' \
+%{endif~}
+%{if var.aws_auto_join~}
+  -retry-join "provider=aws tag_key=Consul-Auto-Join tag_value=${var.name} service=ecs" \
+%{endif~}
+%{if length(var.retry_join_wan) > 0~}
+  -hcl='retry_join_wan = [
+  %{for addr in var.retry_join_wan~}
+    "${addr}",
+  %{endfor~}
+  ]' \
+%{endif~}
+%{if local.enable_mesh_gateway_wan_federation~}
+  -hcl='connect { enable_mesh_gateway_wan_federation = true }' \
+%{endif~}
+%{if length(var.primary_gateways) > 0~}
+  -hcl='primary_gateways = [
+  %{for addr in var.primary_gateways~}
+    "${addr}",
+  %{endfor~}
+  ]' \
+%{endif~}
+EOF
+  // Generate consul_count server configurations.
+  // The index of the range iteration is specified multiple times to be used the same number of times in the template.
+  //consul_server_commands = [for i in range(var.consul_count) : format(local.consul_server_command_template, i)]
+
+  // We use this command to generate the server certs dynamically before the servers start
+  // because we need to add the IP of the task as a SAN to the certificate, and we don't know that
+  // IP ahead of time.
+  consul_server_tls_init_command_template = <<EOF
+ECS_IPV4=$(curl -s $ECS_CONTAINER_METADATA_URI_V4 | jq -r '.Networks[0].IPv4Addresses[0]')
+cd /consul
+echo "$CONSUL_CACERT_PEM" > ./consul-agent-ca.pem
+echo "$CONSUL_CAKEY" > ./consul-agent-ca-key.pem
+consul tls cert create -server \
+  -node="${local.node_name}%s" \
+  -dc="${var.datacenter}" \
+  -additional-ipaddress=$ECS_IPV4 \
+  -additional-dnsname="${var.name}%s.${local.service_discovery_namespace}" \
+%{if length(var.additional_dns_names) > 0~}
+  %{for dnsname in var.additional_dns_names~}
+    -additional-dnsname="${dnsname}" \
+  %{endfor~}
+%{endif~}
+EOF
+
+  consul_portmap = [{
+    containerPort : 8300,
+    hostPort : 8300,
+    protocol : "tcp"
```

Comment:
nit: Please remove the `hostPort` field from these port mappings. We only support the `awsvpc` network mode for our other modules, so we can assume the same here (for now). In `awsvpc` network mode, the [host port cannot be different than the container port](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html), so I'd prefer to leave the field out (similar to the [dev-server module](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/a8cdec6b6ce781679033822acab23f5b58a2129a/modules/dev-server/main.tf#L163-L173))